packages/core/src/StateNode.ts

```diff
@@ -1858,7 +1858,10 @@ class StateNode<
 
     const transition = {
       ...transitionConfig,
-      actions: toActionObjects(toArray(transitionConfig.actions)),
+      actions: toActionObjects(
+        toArray(transitionConfig.actions),
+        this.machine.options.actions
+      ),
```

Comment:
You can't resolve action objects here without breaking existing functionality. I've added a failing test case for this here: https://github.com/statelyai/xstate/commit/272ca85e2529432543197a19f667ed0ccfabe3d5

The same goes for entry & exit actions (those 2 changed lines a little above this one).